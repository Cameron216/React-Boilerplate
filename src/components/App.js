import React from 'react';

import './App.scss';

const App = () => {
    return (
        <div className="app-container">
            <h1 className="app-header">React Boilerplate</h1>
            <p>Project contains:</p>
            <ul>
                <li>Webpack</li>
                <li>React</li>
                <li>Prettier</li>
                <li>Eslint</li>
                <li>Babel</li>
                <li>Autoprefixer</li>
                <li>Jest &amp; Enzyme</li>
            </ul>
        </div>
    );
};

export default App;
