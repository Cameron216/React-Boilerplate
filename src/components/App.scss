@import 'normalize.scss';

$bodyBackground: #cbc887;

body,
html {
    margin: 0;
    padding: 0;
}

body {
    background-color: $bodyBackground;
}

.app-container {
    padding: 1em;
    margin: 0;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;

    .app-header {
        font-size: 2em;
    }
}
