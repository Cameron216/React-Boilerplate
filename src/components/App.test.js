// import React from 'react';
// import renderer from 'react-test-renderer';

// import App from './App';

// describe('App', () => {
//     test('App renders', () => {
//         const component = renderer.create(<App />);
//         let tree = component.toJSON();
//         expect(tree).toMatchSnapshot();
//     });
// });

import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

describe('App', () => {
    it('should render', () => {
        const component = shallow(<App />);

        expect(component).toMatchSnapshot();
    });
});
