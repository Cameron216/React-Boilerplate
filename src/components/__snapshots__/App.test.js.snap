// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<div
  className="app-container"
>
  <h1
    className="app-header"
  >
    React Boilerplate
  </h1>
  <p>
    Project contains:
  </p>
  <ul>
    <li>
      Webpack
    </li>
    <li>
      React
    </li>
    <li>
      Prettier
    </li>
    <li>
      Eslint
    </li>
    <li>
      Babel
    </li>
    <li>
      Autoprefixer
    </li>
    <li>
      Jest & Enzyme
    </li>
  </ul>
</div>
`;
